doc/migration/v5.md

```diff
@@ -0,0 +1,184 @@
+# Angular Migration Guide v4 to v5
+
+## Major changes
+
+- Build system switched from pub transformers to build_runner/webdev. 
+  [Dart 2 Migration Guide for Web Apps](https://webdev-dartlang-org-dev.firebaseapp.com/dart-2)
+- Introduced Router 2.0. [Router Migration Guide](https://github.com/eredo/angular/blob/master/doc/router/migration.md)
+- Strictly typed injector.
+- Usage of generated `.template.dart` files for `NgFactory` referencing.
+
+## Migration steps
+
+Angular 5 and the build related dependencies require Dart 2, therefore install `>=2.0.0-dev.64.1` SDK version before 
+continuing. When developing in Intellij/WebStorm make sure to adapt the configuration of your SDK path in the dart 
+settings.
+
+### Update angular dependencies
+
+```yaml
+dependencies:
+  angular: ^5.0.0-alpha
+  angular_router: ^2.0.0-alpha
+  angular_forms: ^2.0.0-alpha
+
+dev_dependencies:
+  angular_test: ^2.0.0-alpha
+```
+
+### Switch to build_runner
+
+Add or update the build_runner related dependencies to your `pubspec.yaml`.
+
+```yaml
+dev_dependencies:
+  build_runner: ^0.9.0
+  build_test: ^0.10.2
+  build_web_compilers: ^0.4.0
+```
+
+> Notice: It's no longer required to use a separate tool/build.dart in case previous versions of build_runner where
+  used within the project. If builders outside of angular are used, see [build_config README](https://github.com/dart-lang/build/tree/master/build_config) 
+  for instructions to setup an individual build configuration.
+
+Make sure to ignore the `.dart_tool` directory in your `.gitignore` configuration. The directory will contain the
+generated results of the `build_runner`.
+
+> Notice: When developing in Intellij/WebStorm it's currently also recommended to mark the directory as excluded as it 
+  will constantly change and cause Intellij to reanalyze the code frequently, which results in a loss of performance.
+
+Drop transformer related dependencies.
+
+- dart_to_js_script_rewriter
+- browser
+
+Remove transformer configuration from `pubspec.yaml`. 
+
+**Old code:**
+
+```yaml
+dependencies:
+  # ...
+
+dev_dependencies:
+  # ...
+
+transformers:
+- angular:
+    platform_directives:
+    - 'package:angular2/common.dart#COMMON_DIRECTIVES'
+    platform_pipes:
+    - 'package:angular2/common.dart#COMMON_PIPES'
+    entry_points: web/main.dart
+- test/pub_serve:
+    $include:
+      - test/**.dart
+- dart_to_js_script_rewriter
+```
+
+**New code:**
+
+```yaml
+dependencies:
+  # ...
+
+dev_dependencies:
+  # ...
+```
+
+Move your dart2js configuration out of your `pubspec.yaml` into a `build.release.yaml` config. 
+*This step is optional and the configuration may contain different configurations.*
+
+**Old code (`pubspec.yaml`):**
+
+```yaml
+transformers:
+- $dart2js:
+    checked: false
+    minify: true
+    commandLineOptions: [--show-package-warnings]
+```
+
+**New code(`build.release.yaml`):**
```

Comment:
I would drop references to `build.release.yaml` and suggest using `pub run build_runner --release` (or `-r`) instead of `--config release`. We now have built in support for using `dart2js` in "release" mode. Our intention is for most users to not need a `build.yaml` at all.